### PR TITLE
LPS-129705 Implement ImageEditor in DL

### PR DESCRIPTION
**Pull request by carloslancha**
https://issues.liferay.com/browse/LPS-129705

Created a base version of ImageEditor to implement it in DL.

Test plan:

- Go to Documents and Media
- Upload an image
- Got to documents list card view
- Open the document actions dropdown
- Click on Edit Image

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
Just started reviewing :)

**Comment by carloslancha** [comment]
@AliciaGarciaGarcia updated

**Comment by AliciaGarciaGarcia** [comment]
Just started reviewing :)

**Comment by AliciaGarciaGarcia** [comment]
Just started reviewing :)

**Comment by AliciaGarciaGarcia** [comment]
Just started reviewing :)

**Comment by carloslancha** [comment]
@adolfopa @boton please take a look and if lgty feel free to forward it

**Comment by carloslancha** [comment]
ci:test:sf

**Comment by carloslancha** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by liuxingye** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/css/_image_editor.scss [hunk 1]
```diff
@@ -0,0 +1,31 @@
+.image-editor-modal {
+	.modal-body {
+		background-color: $dark;
+		padding: 0;
+	}
+
+	.tbar {
```
Comment (boton) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/css/_image_editor.scss:7]: Is this `.tbar` style always apply inside a modal? 
Maybe `.tbar` fit better outside of `.image-editor-modal` and make `.image-editor` or whatever without modal.

**Review by boton — COMMENTED** [review]
Tested and worked for me, just left a couple of comments there :)

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/css/_image_editor.scss (hunk 1 above)
Comment (carloslancha) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/css/_image_editor.scss:7]: At this point, yes. What you suggest probably will be needed once we implement it in the `item-selector`, but for now this is correct.

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/image-editor/ImageEditor.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+		<>
+			<div style={{height: '100%'}}>
+				<img
+					ref={ref}
+					src={imageSrc}
+					style={{
+						display: 'block',
+						maxWidth: '100%',
+						opacity: 0,
+					}}
```
[the remark this answers is not in the document]
Comment (carloslancha) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/image-editor/ImageEditor.js:176]: This is part of the required API of cropper js

**Comment by carloslancha** [comment]
ci:forward

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/image-editor/ImageEditor.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (boton) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/image-editor/ImageEditor.js:176]: Cool, the `img` styles are API I totally miss it but I suppose that cropper has been expected these values, but the wrapper with `<div style={{height: '100%'}}>` is expected or updated by cropper js or is our styles 🤔

**Comment by boton** [comment]
ci:test:relevant

**Comment by carloslancha** [comment]
ci:stop

**Comment by carloslancha** [comment]
ci:test:sf

**Comment by carloslancha** [comment]
ci:test:relevant

**Comment by carloslancha** [comment]
ci:stop

**Comment by carloslancha** [comment]
ci:forward

**Comment by Hanlf** [comment]
[WorkflowMetrics#ViewReports](https://testray.liferay.com/zh/home/-/testray/cases/1311689211) is an unstable failed test, and not related to this pull, will rerun it.

**Comment by Hanlf** [comment]
ci:forward

**Pull request closed without merging**